tests/test_filters.py

```diff
@@ -2180,6 +2180,24 @@ def test_transform_mesh_and_vectors(datasets, num_cell_arrays, num_point_data):
             )
 
 
+@pytest.mark.parametrize("num_cell_arrays,num_point_data", itertools.product([0, 1, 2], [0, 1, 2]))
+def test_transform_int_vectors_warning(datasets, num_cell_arrays, num_point_data):
```

Comment:
Can we also get a test that makes sure that cell vectors are transformed correctly when not an integer?  It seems the test above this one is broken.  I assume that the transform is overwriting the data in the original mesh, which could be another bug.  I haven't delved too deeply though.

Reply:
Good point, I checked `test_transform_mesh_and_vectors` and indeed the checks are not working. I've determined the `test_transform_mesh_and_vectors` source and it seems to be fine. I can't figure out the cause right now. I think it needs to be investigated.

Reply:
Do we need this to be resolved before reviewing/merging this?

Reply:
Yes. Resolving #2660 entirely is the goal of this PR.

Reply:
> I assume that the transform is overwriting the data in the original mesh, which could be another bug.

Added an additional check for this with 6a223884b80dd72f3c60370a5ba8ac2438fe5e6c.

> Can we also get a test that makes sure that cell vectors are transformed correctly when not an integer?

This is being checked and the test is valid. I like how the test is written to validate a 90 degree rotation.
https://github.com/pyvista/pyvista/blob/6a223884b80dd72f3c60370a5ba8ac2438fe5e6c/tests/test_filters.py#L2199-L2208